dali/pipeline/workspace/workspace.h

```diff
@@ -526,20 +527,92 @@ class WorkspaceBase : public ArgumentWorkspace {
   }
 
   /**
- * @brief Returns the index of the sample that this workspace stores
- * in the input/output batch.
- */
+   * @brief Returns the index of the sample that this workspace stores
+   * in the input/output batch.
+   */
   DLL_PUBLIC virtual inline int data_idx() const {
     return 0;
   }
 
   /**
- * @brief Returns the index of the thread that will process this data.
- */
+   * @brief Returns the index of the thread that will process this data.
+   */
   DLL_PUBLIC virtual inline int thread_idx() const {
     return 0;
   }
 
+
+  /// @{
+  /**
+   * Sets the operator ID that this Workspace belongs to.
+   */
+  void SetOperatorId(std::string operator_id) {
+    operator_id_ = std::move(operator_id);
+  }
+
+
+  /**
+   * Returns the operator ID that this Workspace belongs to.
+   */
+  const std::string &GetOperatorId() const {
+    return operator_id_;
+  }
+  /// @}
+
+
+  ///@{
+  /**
+   * Set the whole operator trace map in this workspace.
+   *
+   * Sets the operator trace map that corresponds to all operators in the current iteration.
```

Comment:
`to all operators`? Isn't this maps specific to the particular operator?

Reply:
`operator_trace_map_t` corresponds to all operators in a given iteration:
```c++
using operator_trace_map_t = std::unordered_map<
        std::string /* op_name */,
        std::unordered_map<std::string /* trace_name */, std::string /* trace_value */>
>;
```
All Workspaces in a given iteration must know all the traces in that iteration. This is stipulated by the output mechanism - output is just another workspace. Therefore, to access any particular op in the iteration, the knowledge of every operator traces is required.